x/wasm-storage/keeper/keeper.go

```diff
@@ -35,139 +78,164 @@ func (k Keeper) GetAuthority() string {
 }
 
 // SetDataRequestWasm stores Data Request Wasm using its hash as the key.
-func (k Keeper) SetDataRequestWasm(ctx sdk.Context, wasm *types.Wasm) {
-	store := ctx.KVStore(k.storeKey)
-	bz := k.cdc.MustMarshal(wasm)
-	store.Set(types.GetDataRequestWasmKey(wasm.Hash), bz)
+func (k Keeper) SetDataRequestWasm(ctx sdk.Context, wasm *types.Wasm) error {
+	dataRequestWasmKeyPrefixFull := GetDataRequestWasmKeyPrefixFull(wasm.Hash)
+
+	return k.DataRequestWasm.Set(ctx, dataRequestWasmKeyPrefixFull, *wasm)
 }
 
 // GetDataRequestWasm returns Data Request Wasm given its key.
-func (k Keeper) GetDataRequestWasm(ctx sdk.Context, hash []byte) *types.Wasm {
-	var wasm types.Wasm
-	store := ctx.KVStore(k.storeKey)
-	bz := store.Get(types.GetDataRequestWasmKey(hash))
-	k.cdc.MustUnmarshal(bz, &wasm)
-	return &wasm
+func (k Keeper) GetDataRequestWasm(ctx sdk.Context, hash []byte) (*types.Wasm, error) {
+	dataRequestWasmKeyPrefixFull := GetDataRequestWasmKeyPrefixFull(hash)
+	wasm, err := k.DataRequestWasm.Get(ctx, dataRequestWasmKeyPrefixFull)
+	if err != nil {
+		return nil, err
+	}
+
+	return &wasm, nil
 }
 
 // HasDataRequestWasm checks if a given Data Request Wasm exists.
-func (k Keeper) HasDataRequestWasm(ctx sdk.Context, wasm *types.Wasm) bool {
-	store := ctx.KVStore(k.storeKey)
-	return store.Has(types.GetDataRequestWasmKey(wasm.Hash))
+func (k Keeper) HasDataRequestWasm(ctx sdk.Context, wasm *types.Wasm) (bool, error) {
+	dataRequestWasmKeyPrefixFull := GetDataRequestWasmKeyPrefixFull(wasm.Hash)
+
+	return k.DataRequestWasm.Has(ctx, dataRequestWasmKeyPrefixFull)
```

Comment:
Can be merged into single line

Reply:
[62cfbab](https://github.com/sedaprotocol/seda-chain/pull/192/commits/62cfbabec5a4ad9788cf1ae06c725cf9b2b39a4b)